certbot/main.py

```diff
@@ -366,6 +367,50 @@ def _init_le_client(config, authenticator, installer):
     return client.Client(config, acc, authenticator, installer, acme=acme)
 
 
+def register(config, unused_plugins):
+    """Create or modify accounts on the server."""
+
+    # Portion of _determine_account logic to see whether accounts already
+    # exist or not.
+    account_storage = account.AccountFileStorage(config)
+    accounts = account_storage.find_all()
+
+    # registering a new account
+    if not config.update_registration:
+        if len(accounts) > 0:
+            # TODO: add a flag to register a duplicate account (this will
+            #       also require extending _determine_account's behavior
+            #       or else extracting the registration code from there)
+            return ("There is an existing account; registration of a "
+                    "duplicate account with this command is currently "
+                    "unsupported.")
+        # _determine_account will register an account
+        _determine_account(config)
+        return
+
+    # --update-registration
+    if len(accounts) == 0:
+        return "Could not find an existing account to update."
+    if config.email is None:
+        return ("Currently, --update-registration can only change the e-mail "
+                "address\nassociated with an account. A new e-mail address is "
+                "required\n(hint: --email)")
+    acc, acme = _determine_account(config)
+    acme_client = client.Client(config, acc, None, None, acme=acme)
+    try:
+        updated_reg = client.messages.Registration.from_data(email=config.email)
+        acme_client.acme.update_registration(acme_client.account.regr,
+                                             updated_reg)
+    except acme_errors.UnexpectedUpdate:
+        # We expect the unexpected update!
+        pass
```

Comment:
The fact we have to catch and raise an exception here indicates to me that we're not going about this the best way. For example,  [this code](https://github.com/certbot/certbot/blob/v0.6.0/acme/acme/client.py#L163) uses `update_registration` without raising an exception.

I played with it a bit and was able to update a registration using this code:
```
acme_client.acme.update_registration(
    acc.regr.update(body=acc.regr.body.update(contact=('mailto:' + config.email,))))
```
While absolutely horrendous, it works and seems to be the way that `update_registration` was designed to work. Additionally, this function returns the updated registration so we don't need the additional call and trip to Boulder with `query_registration`.